testsuite/python/dawaanr-and-dds-gpu.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import print_function
+import unittest as ut
+import numpy as np
+import espressomd
+from espressomd.interactions import *
+from espressomd.magnetostatics import *
+from espressomd.analyze import *
+import math
+from tests_common import *
+from numpy import linalg as la
+from numpy.random import random
+from espressomd import assert_features, has_features, missing_features
+
+if has_features(["DIPOLES","CUDA","PARTIAL_PERIODIC","ROTATION"]):
```

Comment:
Sorry for requesting more changes, but ideally you'd use the `@ut.skipIf` decorator instead of wrapping the whole class definition into an `if` statement, i.e.
````python
@ut.skipIf(not has_features(["DIPOLES","CUDA","PARTIAL_PERIODIC","ROTATION"]),
           "Features not available, skipping test!")
````